CSCD488-Winter2024/senior-project-the-a-team
# Notifications

**Steps to Test:**
- Have 2 people (preferably, or 2 devices to run the app)
- On both devices, checkout the Notifications Branch and run flutter pub get
- On one, login, and then verify on firebase that the account's notification token if filled
- Also, a popup requesting notification permissions should appear.
- Then, minimize the app on that first one
- On the second, preferably an admin/poster, create an alert post, and verify that a notification is received on the first device. It can take about 15 seconds
- Repeat this step for:
      a post that has a tag that the first account is interested in (should receive notif)
      a post that doesnt have a tag that the first account is interested in (shouldn't)
      an event that has the tag (should)
      an event that doesnt (shouldn't)
      a job (shouldn't)
      a volunteer (shouldn't)

**Notes:**
- index.js has the cloud code
- main.dart asks the permissions
- login.dart stores the notif token
- register.dart defaults the user's notificationToken field to "none"
- the rest is all just misc dependencies
